pkg/sql/parser/sql.y

```diff
@@ -3722,7 +3722,14 @@ typename:
   }
   // SQL standard syntax, currently only one-dimensional
 | simple_typename ARRAY '[' ICONST ']' { return unimplementedWithIssue(sqllex, 2115) }
```

Comment:
It seems to me that in the same way we support `VARCHAR(n)` and ignore `n`, we could support `ARRAY[n]` and ignore `n` too.